CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+#  Contribution Guidelines and Policies
+
+## Contribution Guidelines
+
+We encourage contributions to the NetworKit source code.
+If you encounter any questions regarding NetworKit development open an issue on GitHub or contact the [mailing list][list].
+
+### General Guidelines
+
+- NetworKit comes with a large collection of unit tests.
+  Please provide tests for new code and for behavior that is modified.
+  See below for instructions on building and running the tests.
+- We generally welcome Jupyter notebooks that demonstrate how new algorithms are used. If you add a new feature,
+  consider adding a notebook to the `notebooks/` directory (or add your feature to an existing notebook).
+
+### Coding Style
+
+- We use 4 space indentation.
+- Adapt to the existing naming conventions. NetworKit generally uses lowerCamelCase
+  for (global and local) variables and UpperCamelCase for classes.
+- We have a `.clang-format` file in the root directory. We do not require all code to follow the coding style specified
+  in this file. However, it should be taken as a suggestion.
```

Comment:
We might suggest to add `// networkit-format` on the files that are clang-formatted.